lib/extras/enc/jpg.cc

```diff
@@ -235,5 +233,90 @@ Status EncodeImageJPG(const CodecInOut* io, JpegEncoder encoder, size_t quality,
   return true;
 }
 
+Status ParseChromaSubsampling(const std::string& param,
+                              YCbCrChromaSubsampling* subsampling) {
+  constexpr std::pair<const char*,
+                      std::pair<std::array<uint8_t, 3>, std::array<uint8_t, 3>>>
+      options[] = {{"444", {{{1, 1, 1}}, {{1, 1, 1}}}},
+                   {"420", {{{2, 1, 1}}, {{2, 1, 1}}}},
+                   {"422", {{{2, 1, 1}}, {{1, 1, 1}}}},
+                   {"440", {{{1, 1, 1}}, {{2, 1, 1}}}}};
+  for (const auto& option : options) {
+    if (param == option.first) {
+      return subsampling->Set(option.second.first.data(),
+                              option.second.second.data());
+    }
+  }
+  return false;
+}
+
+class JPEGEncoder : public Encoder {
+  std::vector<JxlPixelFormat> AcceptedFormats() const override {
+    std::vector<JxlPixelFormat> formats;
+    for (const uint32_t num_channels : {1, 3}) {
```

Comment:
FWIW, this one _does_ support CMYK, although I guess we don't really have a way to express that.

Reply:
Well, CMYK JPEGs exist, although how they get created I do not know :)

Reply:
We don't currently support reading nor writing of CMYK jpegs in that code, but it could be added, it shouldn't be too hard to add. Just can only decode them to pixels and encode them from pixels, not dct coeffs, since VarDCT has 3 channels.

(theoretically we could represent the CMY losslessly as dct coeffs, and only convert the K to pixels, but that would require something other than PackedPixelFiles and I don't think it's worth the trouble).

Reply:
I think it makes sense to implement CMYK encoding later and add it to `AcceptedFormats` at that point, but in the meantime, it would seem strange for the encoder to advertise support for it when it is not actually implemented.